content/docs/0.16.x/reference/files/shipyard/index.md

```diff
@@ -77,8 +77,20 @@ although most projects only use some of the constructions.
 A stage is named for the particular activity to be performed,
 such as `development`, `hardening`, `staging`, or `production`.
 Each *shipyard* file must have at least one `stage`.
-The name of the stage becomes the name of the branch in the upstream Git repository
-and the Kubernetes namespace to which services are deployed.
+The name of the stage becomes the name of the branch
+in the [upstream Git repository](../../../manage/git_upstream)
+and the Kubernetes namespace to which
+[services](../../../manage/service) are deployed.
+
+A stage can be given any meaningful name that conforms to the
+[Kubernetes Object Names and IDs](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/)
+specification, meaning:
+
+* contains at most 63 characters
```

Comment:
Keptn reduces this value to 43: https://github.com/keptn/keptn/blob/master/installer/manifests/keptn/values.yaml#L246
However, users can change this number via the Helm value `shipyardController.config.serviceNameMaxSize`

Reply:
Interesting, thanks!  Is there a limit to how high this value can be set -- like the k8s limit?  Are there good reasons to not increase the limit?  We may as well be thorough here, right?

Reply:
I don't recall why we selected 43 as a magic number, I'll check with the team and let you know :)

Reply:
Due to DNS limitations, Kubernetes allows only 63 char names. For the continuous delivery use case, the `helm-service` creates Helm releases, which have [53 char limits](https://helm.sh/docs/chart_template_guide/getting_started/#adding-a-simple-template-call). Keptn makes a release name with the following format: `<serviceName>-generated`. Since `-generated` is 10char long, we limit the `serviceName` part to be 43chars.

I would say that this value can increase if the user does not use the `helm-service`.

Reply:
Fabulous!  I updated the content with this new information; can update further if necessary.